src/go/rpk/pkg/cli/cloud/mcp/mcp.go

```diff
@@ -209,3 +197,314 @@ func newStdioCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+
+This command connects to a specific cluster and MCP server running in that cluster,
+then proxies MCP requests from stdio to the remote MCP server over HTTP.
+
+Use --install to configure the MCP client instead of serving stdio.`,
+		Args: cobra.NoArgs,
+		PreRunE: func(_ *cobra.Command, _ []string) error {
+			if install && mcpClient == "" {
+				return fmt.Errorf("--client flag is required when using --install")
+			}
+			return nil
+		},
+		Run: func(_ *cobra.Command, _ []string) {
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "Failed to load config: %w", err)
+
+			// Start out with empty token, and use the maybeReloadToken function to update.
+			cl := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, "")
+			var m sync.RWMutex
+			var tokenOK bool
+
+			maybeReloadToken := func() error {
+				cfg, err := p.Load(fs)
+				if err != nil {
+					return err
+				}
+
+				var authVir *config.RpkCloudAuth
+				yVir := cfg.VirtualRpkYaml()
+				pVir := yVir.Profile(yVir.CurrentProfile)
+				authVir = pVir.VirtualAuth()
+				if authVir == nil {
+					authVir = yVir.CurrentAuth() // must be non-nil; we always have a default virtual auth
+				}
+
+				extra := ""
+				if e := cfg.VirtualProfile().CloudEnvironment; e != "" {
+					extra = fmt.Sprintf(" -X cloud_environment=%s", e)
+				}
+				authToken := authVir.AuthToken
+				acl := auth0.NewClient(cfg.DevOverrides())
+				expired, err := oauth.ValidateToken(authToken, acl.Audience(), authVir.ClientID)
+				if err != nil {
+					m.Lock()
+					tokenOK = false
+					m.Unlock()
+					return fmt.Errorf("failed to validate Cloud token. This may happen, if they never logged in to Redpand Cloud. However, their token is not just expired, we'd not see this error then. Instruct the user to run `rpk%s cloud login --no-profile` to login to Redpanda Cloud. After running the command, they can ask to retry", extra)
+				}
+				if expired {
+					m.Lock()
+					tokenOK = false
+					m.Unlock()
+					return fmt.Errorf("the Redpanda Cloud token is expired. Instruct the user to run `rpk%s cloud login --no-profile` to obtain a fresh one. Afterwards, they can ask to retry", extra)
+				}
+				m.RLock()
+				ok := tokenOK
+				m.RUnlock()
+				if !ok {
+					cl.UpdateAuthToken(authToken)
+					m.Lock()
+					tokenOK = true
+					m.Unlock()
+				}
+				return nil
+			}
+
+			// Load token initially
+			err = maybeReloadToken()
+			out.MaybeDie(err, "Failed to load authentication token: %v", err)
+
+			// Get the auth token for dataplane client
+			yVir := cfg.VirtualRpkYaml()
+			pVir := yVir.Profile(yVir.CurrentProfile)
+			authVir := pVir.VirtualAuth()
+			if authVir == nil {
+				authVir = yVir.CurrentAuth() // must be non-nil; we always have a default virtual auth
+			}
+			authToken := authVir.AuthToken
+
+			// Get cluster information
+			cluster, err := cl.Cluster.GetCluster(context.Background(), connect.NewRequest(&controlplanev1.GetClusterRequest{
```

Comment:
For all the context usage in the command to use the underlying cmd context
```suggestion
			cluster, err := cl.Cluster.GetCluster(cmd.Context(), connect.NewRequest(&controlplanev1.GetClusterRequest{
```

And in L221 replace `_` with `cmd`